File: increasingring/src/main/java/com/baybaka/increasingring/RunTimeSettings.java
package com.baybaka.increasingring;

import android.app.Notification;
import android.content.Context;

public interface RunTimeSettings {

    boolean isLoggingEnabled();

    void setListenerIsRegistered();

    void setListenerUnregistered();

    void configurationChanged();

    void configIsUpdated();

    boolean isConfigChanged();

    Context getContext();

    Notification persistentNotification(Context context, boolean minPriority);

    void errorNotification(Context context);

    void findPhoneNotification();

    /**
     *
     * @return is service running and listener registered
     */
    boolean isPhoneStateListenerRegistered();

    boolean isTestPageOpened();

    void setTestPageOpened(boolean state);

    void setGlobalLoggingState(boolean state);


    //to del
    long getLastVolActionTime();

    void setLastVolActionTime(long mils);
}

File: increasingring/src/main/java/com/baybaka/increasingring/settings/RunTimeSettings.kt
package com.baybaka.increasingring.settings

import android.app.Notification
import android.content.Context
import com.baybaka.increasingring.interfaces.NotificationProvider

interface RunTimeSettings {

    val isLoggingEnabled: Boolean

    fun setListenerIsRegistered()

    fun setListenerUnregistered()

    fun configurationChanged()

    fun configIsUpdated()

    val isConfigChanged: Boolean

    val context: Context

    //todo notification should be move out
    fun persistentNotification(context: Context, minPriority: Boolean): Notification

    fun errorNotification(context: Context)

    fun findPhoneNotification()

    /**

     * @return is service running and listener registered
     */
    val isPhoneStateListenerRegistered: Boolean

    var isTestPageOpened: Boolean

    fun setGlobalLoggingState(state: Boolean)

    fun getNotifyProvider() : NotificationProvider
}
